src/frontend/src/optimizer/plan_node/generic.rs

```diff
@@ -832,10 +837,11 @@ pub struct Filter<PlanRef> {
     pub predicate: Condition,
     pub input: PlanRef,
 }
+impl_plan_tree_node_v2_for_generic_unary_node!(Filter, input);
 
 /// `TopN` sorts the input data and fetches up to `limit` rows from `offset`
 #[derive(Debug, Clone)]
-pub struct TopN<PlanRef> {
+pub struct TopN<PlanRef: Clone> {
```

Comment:
I don't think you need this constraint. You can put it in the impl but here it's unnecessary